frontend/src/entry/model/InferenceServices.tsx

```diff
@@ -35,17 +98,35 @@ export default function InferenceServices({ model }: InferenceProps) {
     return <MessageAlert message={isInferencesError.info.message} severity='error' />
   }
 
+  if (isTokensLoading || isUiConfigLoading) {
+    return <Loading />
+  }
+
   return (
     <Container sx={{ my: 2 }}>
-      <Stack spacing={4}>
-        <Box sx={{ textAlign: 'right' }}>
-          <Button variant='outlined' onClick={handleCreateNewInferenceService}>
-            Create Service
-          </Button>
-        </Box>
-        {isInferencesLoading && <Loading />}
-        {inferenceDisplays}
-      </Stack>
+      {healthCheck ? (
+        <Stack spacing={4}>
+          <Box sx={{ textAlign: 'right' }}>
+            <Button variant='outlined' onClick={handleCreateNewInferenceService}>
+              Create Service
+            </Button>
+          </Box>
+          {isInferencesLoading && <Loading />}
+          {inferenceDisplays}
+        </Stack>
+      ) : (
+        <Stack spacing={2}>
+          <Typography>
+            Access to inferencing services requires token a with model access. Are you sure you want to proceed?
+          </Typography>
+          <Stack spacing={2} direction='row'>
+            <Button variant='contained' onClick={handleCreateToken}>
+              Yes
+            </Button>
+          </Stack>
+          {errorMessage && <MessageAlert message={errorMessage} severity='error' />}
```

Comment:
`MessageAlert` internally checks `if (!message) return null`, so you don't need to conditionally render it.

I'd also recommend pulling this outside of your conditional logic (move it to just above the </Container> a couple lines below) so that it's always shown if there is an error to be displayed. (Not an issue for what you've done, but someone could easily miss that it's in a conditional block when adding future error handling)